src/Symfony/Component/Security/Http/Authenticator/RememberMeAuthenticator.php

```diff
@@ -57,32 +65,37 @@ public function supports(Request $request): ?bool
             return false;
         }
 
-        // if the attribute is set, this is a lazy firewall. The previous
-        // support call already indicated support, so return null and avoid
-        // recreating the cookie
-        if ($request->attributes->has('_remember_me_token')) {
-            return null;
+        if (($cookie = $request->attributes->get(ResponseListener::COOKIE_ATTR_NAME)) && null === $cookie->getValue()) {
+            return false;
         }
 
-        $token = $this->rememberMeServices->autoLogin($request);
-        if (null === $token) {
+        if (!$request->cookies->has($this->cookieName)) {
             return false;
         }
 
-        $request->attributes->set('_remember_me_token', $token);
+        if (null !== $this->logger) {
+            $this->logger->debug('Remember-me cookie detected.');
+        }
 
         // the `null` return value indicates that this authenticator supports lazy firewalls
         return null;
     }
 
     public function authenticate(Request $request): PassportInterface
     {
-        $token = $request->attributes->get('_remember_me_token');
-        if (null === $token) {
-            throw new \LogicException('No remember me token is set.');
+        $rawCookie = $request->attributes->get(ResponseListener::COOKIE_ATTR_NAME);
+        if (!$rawCookie || null === $rawCookie->getValue()) {
+            $rawCookie = $request->cookies->get($this->cookieName);
```

Comment:
I'm not quite following the logic here. We look for the cookie in `$request->attributes` first and THEN in `$request->cookies`. Why? The flow may just need some clues with short comments

Reply:
I'm not sure if I did this with a reason, or if this is just because that's also how it's done in the "legacy" remember me: https://github.com/symfony/symfony/blob/9abeb248a58d22d94f18c1055f648fdf1a4c80e5/src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php#L101-L107

Reply:
Turns out that this is great catch. It's based on a wrong copy-paste from the services to the authenticator. `authenticate()` should only use the cookie (the null attribute case is covered by `support()` and is required for scenarios as described in https://github.com/symfony/symfony/issues/35198 )